docs/src/main/java/io/quarkus/docs/generation/CopyJavaExamples.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.docs.generation;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CopyJavaExamples {
+
+    public Path outputPath;
+    public List<Path> paths;
+
+    // ${project.basedir}/../target/asciidoc/generated/examples ${project.basedir}/../integration-tests
+    public static void main(String[] args) throws Exception {
+        if (args.length < 2) {
+            System.err.println("Must specify target output directory (first) followed by at least one source directory");
+            System.exit(1);
+        }
+        CopyJavaExamples javaExamples = new CopyJavaExamples();
+
+        javaExamples.outputPath = Path.of(args[0]);
+        javaExamples.paths = Arrays.stream(args).skip(1)
+                .map(Path::of)
+                .collect(Collectors.toList());
+
+        try {
+            javaExamples.run();
+        } catch (Exception e) {
+            System.err.println("Exception occurred while trying to copy java examples");
+            e.printStackTrace();
+            System.exit(1);
+        }
+    }
+
+    public void run() throws Exception {
+        if (outputPath != null) {
+            Files.createDirectories(outputPath);
+        }
+
+        for (Path path : paths) {
+            Files.walkFileTree(path, new SimpleFileVisitor<>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    if (file.toString().contains("src/main/java/documentation/example/")) {
+                        String name = file.toString();
+                        name = name.substring(name.indexOf("example/") + 8)
+                                .replace("/", "-");
+
+                        Files.copy(file, outputPath.resolve(name), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
so ... if two modules uses the same subdir name under example it will just end up overwriting it?

could we detect/avoid it?

if done as I thought about this we would have a module/dir per example with sources in it avoiding at least this particular problem.

Reply:
Yes, I can detect/avoid it.